File: src/main/java/kdh/boardproject/dto/user/UserRequestDto.java
package kdh.boardproject.dto.user;

import kdh.boardproject.entity.User;
import lombok.AllArgsConstructor;
import lombok.Data;

@Data
public class UserRequestDto {
    private Long idx;
    private String id;
    private String name;

    public UserRequestDto(User user){
        this.idx = user.getIdx();
        this.id = user.getId();
        this.name = user.getName();
    }

}

File: src/main/java/kdh/boardproject/dto/user/UserRequestDto.kt
package kdh.boardproject.dto.user

import kdh.boardproject.entity.User
import lombok.Data

@Data
class UserRequestDto(user: User) {
     var idx: Long
     var id: String
     var name: String

    init {
        idx = user.idx!!
        id = user.id
        name = user.name
    }
}